src/ctap/mod.rs

```diff
@@ -2320,7 +2320,8 @@ mod test {
         let mut rng = ThreadRng256 {};
         let key_agreement_key = crypto::ecdh::SecKey::gensk(&mut rng);
         let pin_uv_auth_token = [0x88; 32];
-        let client_pin = ClientPin::new_test(key_agreement_key, pin_uv_auth_token);
+        let client_pin =
+            ClientPin::new_test(key_agreement_key, pin_uv_auth_token, PinUvAuthProtocol::V1);
```

Comment:
I don't see any test using `ClientPin::new_test(..., ..., PinUvAuthProtocol::V2)`.
Is it WAI?

Reply:
> Especially the unit test rework makes this PR huge, so all required changes for V2 outside of the main ClientPin command will be in the next PR.

Tests in `client_pin.rs` are general, everything else is unchanged as much as possible.